File: TodayFitComplete_Backend/src/main/java/com/SJY/TodayFitComplete_Backend/dto/response/board/BoardListResponse.java
package com.SJY.TodayFitComplete_Backend.dto.response.board;

import com.SJY.TodayFitComplete_Backend.entity.Board;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 게시글 목록 응답을 위한 DTO. 양방향 관계로 인해 JSON 직렬화 루프를 피하기 위함.
 */
@Getter
@Setter
@NoArgsConstructor
public class BoardListResponse {

    private Long boardId;
    private String title;
    private String content;
    private int viewCount;
    private String createdDate;
    private String modifiedDate;
    private String writerName;

    @Builder
    public BoardListResponse(Long boardId, String title, String content, int viewCount, String createdDate, String modifiedDate, String writerName) {
        this.boardId = boardId;
        this.title = title;
        this.content = content;
        this.viewCount = viewCount;
        this.createdDate = createdDate;
        this.modifiedDate = modifiedDate;
        this.writerName = writerName;
    }

    public static BoardListResponse fromEntity(Board board) {
        return BoardListResponse.builder()
                .boardId(board.getId())
                .title(board.getTitle())
                .content(board.getContent())
                .viewCount(board.getViewCount())
                .createdDate(board.getCreatedDate())
                .modifiedDate(board.getModifiedDate())
                .writerName(board.getMember().getUsername())
                .build();
    }
}

File: TodayFitComplete_Frontend/app/src/main/java/kr/ac/kumoh/ce/s20190633/todayfitcomplete_frontend/Dto/Board/BoardListResponse.kt
package kr.ac.kumoh.ce.s20190633.todayfitcomplete_frontend.Dto.Board

// BoardListResponse 클래스: 게시글 목록에서 각 게시글의 요약 정보를 나타냅니다.
data class BoardListResponse(
    val boardId: Long,                   // 게시글 고유 ID
    val title: String,                   // 게시글 제목
    val content: String,                 // 게시글 내용
    val viewCount: Int,                  // 게시글 조회수
    val createdDate: String,             // 작성일
    val modifiedDate: String,            // 수정일
    val writerName: String               // 작성자 이름
)